app/javascript/packages/document-capture/hooks/use-steps.ts

```diff
@@ -0,0 +1,94 @@
+import { useContext } from 'react';
+import type { FormStep } from '@18f/identity-form-steps';
+import { useI18n } from '@18f/identity-react-i18n';
+import { UploadFormEntriesError } from '../services/upload';
+import DocumentsStep from '../components/documents-step';
+import InPersonPrepareStep from '../components/in-person-prepare-step';
+import InPersonLocationPostOfficeSearchStep from '../components/in-person-location-post-office-search-step';
+import InPersonLocationFullAddressEntryPostOfficeSearchStep from '../components/in-person-location-full-address-entry-post-office-search-step';
+import InPersonSwitchBackStep from '../components/in-person-switch-back-step';
+import ReviewIssuesStep from '../components/review-issues-step';
+import withProps from '../higher-order/with-props';
+import { InPersonContext } from '../context';
+import UploadContext from '../context/upload';
+
+const useReviewStep = (submissionError: Error | undefined) => {
+  const { t } = useI18n();
+  return {
+    name: 'review',
+    form:
+      submissionError instanceof UploadFormEntriesError
+        ? withProps({
+            remainingAttempts: submissionError.remainingAttempts,
+            isFailedResult: submissionError.isFailedResult,
+            isFailedDocType: submissionError.isFailedDocType,
+            captureHints: submissionError.hints,
+            pii: submissionError.pii,
+            failedImageFingerprints: submissionError.failed_image_fingerprints,
+          })(ReviewIssuesStep)
+        : ReviewIssuesStep,
+    title: t('errors.doc_auth.rate_limited_heading'),
+  };
+};
+
+export const useSteps = (submissionError: Error | undefined) => {
+  const { t } = useI18n();
+  const { inPersonFullAddressEntryEnabled, inPersonURL } = useContext(InPersonContext);
+  const { flowPath } = useContext(UploadContext);
+
+  // getReviewStep needs to be called even if we're not going to use it because it contains
+  // a hook (in a context). Conditionally calling a hook causes problems in React, so
+  // always calling getReviewStep is necessary
+  const reviewStep: FormStep = useReviewStep(submissionError);
```

Comment:
Like components, I'd expect a "one hook per file" rule. What's the downside of just assigning the value generated in `useReviewStep` inline here, aside from noise? Or better yet, we can assign it further down the file past the conditional early return (i.e. alongside `locationStep` and `inPersonProofingSteps`).